staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -44,30 +44,53 @@ import (
 type KeyValidation func(ctx context.Context, t *testing.T, key string)
 
 func RunTestCreate(ctx context.Context, t *testing.T, store storage.Interface, validation KeyValidation) {
-	out := &example.Pod{}
-	obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns", SelfLink: "testlink"}}
+	tests := []struct {
+		name        string
+		inputObj    *example.Pod
+		expectedErr string
+	}{{
+		name:     "normal create",
+		inputObj: &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns", SelfLink: "testlink"}},
+	}, {
+		name:        "create with ResourceVersion set",
+		inputObj:    &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "bar", Namespace: "test-ns", ResourceVersion: "1"}},
+		expectedErr: "resourceVersion should not be set on objects to be created",
```

Comment:
I would extract `errors.New("resourceVersion should not be set on objects to be created")` from the main codepath into an accessible variable so that we can use `errors.Is` for error comparison. It's a little bit cleaner than the direct string comparison, imo.